yjit/src/codegen.rs

```diff
@@ -4702,6 +4702,32 @@ fn jit_rb_str_bytesize(
     true
 }
 
+fn jit_rb_str_getbyte(
+    _jit: &mut JITState,
+    asm: &mut Assembler,
+    _ocb: &mut OutlinedCb,
+    _ci: *const rb_callinfo,
+    _cme: *const rb_callable_method_entry_t,
+    _block: Option<BlockHandler>,
+    _argc: i32,
+    _known_recv_class: *const VALUE,
+) -> bool {
+    asm.comment("String#getbyte");
+    extern "C" {
+        fn rb_str_getbyte(str: VALUE, index: VALUE) -> VALUE;
+    }
+    // rb_str_getbyte() never allocates
```

Comment:
It for non int indexes it calls to_int() so you need to jit_prepare_routine_call().

```
$ ruby -e 'a = []; def a.to_int = p(100); "".getbyte(a)'
100
```